pilot/cmd/pilot-agent/main.go

```diff
@@ -450,6 +470,47 @@ func getDNSDomain(domain string) string {
 	return domain
 }
 
+// check if SDS UDS path and token path exist, if both exist, requests key/cert
+// using SDS instead of secret mount.
+func detectSds(controlPlaneBootstrap, controlPlaneAuthEnabled bool, udspath, preferJwtpath, jwtpath string) (bool, string) {
+	if !controlPlaneBootstrap {
```

Comment:
See my comments, let's avoid this complexity. 

If SDS and secrets are both missing - it's the job of readiness to restart until one is found.

Reply:
Maybe just call it 'legacyCP' - so people are not confused ( new control plane uses a sidecar with same template config as sidecars, no longer uses the deprecated code )

Reply:
not critical, you can leave it as is.

Reply:
thanks.